src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1148,6 +1149,7 @@ private static DocumentTombstone TableValueToTombstone(JsonOperationContext cont
                     }
                     table.Delete(doc.StorageId);
 
+                    // TODO: Do not delete the attachment streams if versioning is on
```

Comment:
I think that we'll not need to. We're creating the versions on Put, so we'll just handle it normally (if there are no other refs to the stream, we'll delete it.